core/services/offchainreporting/oracle.go

```diff
@@ -0,0 +1,198 @@
+	logBroadcaster eth.LogBroadcaster,
+) {
+	jobSpawner.RegisterDelegate(
+		NewJobSpawnerDelegate(db, config, keyStore, pipelineRunner, ethClient, logBroadcaster),
+	)
+}
+
+type jobSpawnerDelegate struct {
+	db             *gorm.DB
+	config         *orm.Config
+	keyStore       *KeyStore
+	pipelineRunner pipeline.Runner
+	ethClient      eth.Client
+	logBroadcaster eth.LogBroadcaster
+}
+
+func NewJobSpawnerDelegate(
+	db *gorm.DB,
+	config *orm.Config,
+	keyStore *KeyStore,
+	pipelineRunner pipeline.Runner,
+	ethClient eth.Client,
+	logBroadcaster eth.LogBroadcaster,
+) *jobSpawnerDelegate {
+	return &jobSpawnerDelegate{db, config, keyStore, pipelineRunner, ethClient, logBroadcaster}
+}
+
+func (d jobSpawnerDelegate) JobType() job.Type {
+	return JobType
+}
+
+func (d jobSpawnerDelegate) ToDBRow(spec job.Spec) models.JobSpecV2 {
+	concreteSpec, ok := spec.(OracleSpec)
+	if !ok {
+		panic(fmt.Sprintf("expected an offchainreporting.OracleSpec, got %T", spec))
+	}
+	return models.JobSpecV2{OffchainreportingOracleSpec: &concreteSpec.OffchainReportingOracleSpec}
+}
+
+func (d jobSpawnerDelegate) FromDBRow(spec models.JobSpecV2) job.Spec {
+	if spec.OffchainreportingOracleSpec == nil {
+		return nil
+	}
+	return &OracleSpec{
+		OffchainReportingOracleSpec: *spec.OffchainreportingOracleSpec,
+		jobID:                       spec.ID,
+	}
+}
+
+func (d jobSpawnerDelegate) ServicesForSpec(spec job.Spec) ([]job.Service, error) {
+	concreteSpec := spec.(*OracleSpec)
+
+	aggregator, err := ocrcontracts.NewOffchainReportingAggregator(
+		concreteSpec.ContractAddress,
+		d.ethClient,
+		d.logBroadcaster,
+		concreteSpec.JobID(),
+		nil, // auth *bind.TransactOpts,
+		"",  // rpcURL string,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	p2pkey, exists := d.keyStore.DecryptedP2PKey(peer.ID(concreteSpec.P2PPeerID))
+	if !exists {
+		return nil, errors.Errorf("P2P key '%v' does not exist", concreteSpec.P2PPeerID)
+	}
+
+	ocrkey, exists := d.keyStore.DecryptedOCRKey(concreteSpec.EncryptedOCRKeyBundleID)
+	if !exists {
+		return nil, errors.Errorf("OCR key '%v' does not exist", concreteSpec.EncryptedOCRKeyBundleID)
+	}
+
+	logger := logger.NewOCRLogger(logger.Default)
+
+	peer, err := ocrnetworking.NewPeer(ocrnetworking.PeerConfig{
+		PrivKey:    p2pkey.PrivKey,
+		ListenPort: d.config.OCRListenPort(),
+		ListenIP:   d.config.OCRListenIP(),
+		Logger:     logger,
+		Peerstore:  pstoremem.NewPeerstore(),
+		EndpointConfig: ocrnetworking.EndpointConfig{
+			IncomingMessageBufferSize: d.config.OCRIncomingMessageBufferSize(),
+			OutgoingMessageBufferSize: d.config.OCROutgoingMessageBufferSize(),
+			NewStreamTimeout:          d.config.OCRNewStreamTimeout(),
+			DHTLookupInterval:         d.config.OCRDHTLookupInterval(),
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	oracle, err := ocr.NewOracle(ocr.OracleArgs{
+		LocalConfig: ocrtypes.LocalConfig{
+			DataSourceTimeout:                      time.Duration(concreteSpec.ObservationTimeout),
```

Comment:
Why not just make it a `Duration` type on the spec instead of casting here?

Reply:
We need custom `.Unmarshal()`/`.Scan()`/`.Value()` logic.  Same reason `models.Duration` exists

Reply:
Right OK. This is a bit boilerplatey but I can live with it since we'd need to cast `models.Duration` anyway.